operator-framework-core/src/main/java/io/javaoperatorsdk/operator/ReconcilerUtils.java

```diff
@@ -98,4 +103,47 @@ public static String getDefaultReconcilerName(String reconcilerClassName) {
     }
     return reconcilerClassName.toLowerCase(Locale.ROOT);
   }
+
+  public static void handleKubernetesClientException(Exception e, String resourceTypeName) {
+    if (e instanceof MissingCRDException) {
+      throw ((MissingCRDException) e);
+    }
+
+    if (e instanceof KubernetesClientException) {
+      KubernetesClientException ke = (KubernetesClientException) e;
+      if (404 == ke.getCode()) {
+        // only throw MissingCRDException if the 404 error occurs on the target CRD
+        if (resourceTypeName.equals(ke.getFullResourceName())
+            || matchesResourceType(resourceTypeName, ke)) {
+          throw new MissingCRDException(resourceTypeName, null, e.getMessage(), e);
+        }
+      }
+    }
+  }
+
+  private static boolean matchesResourceType(String resourceTypeName,
+      KubernetesClientException exception) {
+    final var fullResourceName = exception.getFullResourceName();
+    if (fullResourceName != null) {
+      return resourceTypeName.equals(fullResourceName);
+    } else {
+      // extract matching information from URI in the message if available
+      final var message = exception.getMessage();
+      final var regex = Pattern.compile(".*http(s?)://[^/]*/api(s?)/(\\S*).*").matcher(message);
```

Comment:
We should do something about the related SONAR issue (maybe just no sonar?) or based on the instructions.